src/ansys/fluent/core/quantity.py

```diff
@@ -1,1369 +1,830 @@
-                (
-                    conversion_factor,
-                    unit_str,
-                ) = _UnitsTable.derived_units_with_conversion_factor[unit_str]
-                self._si_multiplier *= conversion_factor**term_power
-                self._compute_multipliers(unit_str, term_power)
-            elif unit_str in _UnitsTable.derived_units:
-                self._compute_multipliers(
-                    _UnitsTable.derived_units[unit_str], term_power
-                )
+    def __rtruediv__(self, __value):
+        return self.__truediv__(__value)
+
+    def __add__(self, __value):
+        new_unit_str = self._arithmetic_precheck(__value)
+        new_value = float(self) + float(__value)
+        return Quantity(value=new_value, unit_str=new_unit_str)
+
+    def __radd__(self, __value):
+        return self.__add__(__value)
+
+    def __sub__(self, __value):
+        new_unit_str = self._arithmetic_precheck(__value)
+        new_value = float(self) - float(__value)
+        return Quantity(value=new_value, unit_str=new_unit_str)
 
-    @staticmethod
-    def _power_sum(base, unit_str):
+    def __rsub__(self, __value):
+        return self.__sub__(__value)
+
+    def __neg__(self):
+        return Quantity(-self.value, self.unit_str)
+
+    def __gt__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) > float(__value)
+
+    def __ge__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) >= float(__value)
+
+    def __lt__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) < float(__value)
+
+    def __le__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) <= float(__value)
+
+    def __eq__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) == float(__value)
+
+    def __neq__(self, __value):
+        self._arithmetic_precheck(__value)
+        return float(self) != float(__value)
+
+
+class QuantityError(ValueError):
+    """Custom quantity errors."""
+
+    def __init__(self, from_unit, to_unit):
         """
-        Sum of powers.
+        Quantity errors with custom messages.
 
         Parameters
         ----------
-        base: str
-            Unit term
-        unit_str: str
+        from_unit: str
             Unit of quantity
+        to_unit: str
+            Desired conversion unit
+        """
+        self.from_unit = from_unit
+        self.to_unit = to_unit
+
+    def __str__(self):
+        return f"'{self.from_unit}' and '{self.to_unit}' have incompatible dimensions."
+
+
+class UnitsTable(object):
+    """Initializes a UnitsTable object with all table values and unit string manipluation methods.
+
+    Properties
+    ----------
+    api_quantity_map : dict
+        Quantity map values from settings API.
+    fundamental_units : dict
+        Fundamental units and properties.
+    derived_units : dict
```

Comment:
Same here for doc.  Something like "Common units that are derived from fundamental units".